src/Microsoft.DotNet.Wpf/tests/UnitTests/Directory.Build.targets

```diff
@@ -24,15 +23,19 @@
   </Target>
 
   <!--
-    A WindowsBase facade is included in the Microsoft.NETCore.App targeting pack while we ship our own WindowsBase in the Microsoft.WindowsDesktop.App targeting pack.
-    To allow our projects to reference our version of WindowsBase without conflicting, we remove the version from the Microsoft.NETCore.App targeting pack from the list of reference.
+   Referencing packages that reference the desktop package causes grief with building internally. Currently this
+   is only happening with referencing StaFact (for [WpfFact] tests). Remove the reference if it comes in directly
+   or transitively.
   -->
-  <Target Name="RemoveWindowsBaseNetCoreAppReference"
-          AfterTargets="ResolveTargetingPackAssets"
-          Returns="@(Reference)">
+  <Target Name="RemoveDesktopFramework" AfterTargets="ProcessFrameworkReferences" BeforeTargets="FixupRuntimeFrameworkVersions">
     <ItemGroup>
-      <Reference Remove="@(Reference)" Condition="'%(Reference.AssemblyName)' == 'WindowsBase' And '%(Reference.FrameworkReferenceName)' == 'Microsoft.NETCore.App'" />
+      <RuntimeFramework Remove="@(RuntimeFramework)" Condition="'%(RuntimeFramework.Identity)' == 'Microsoft.WindowsDesktop.App'"/>
     </ItemGroup>
   </Target>
 
+  <Target Name="RemoveTransitiveDesktopFramework" AfterTargets="ResolvePackageAssets" BeforeTargets="AddTransitiveFrameworkReferences">
```

Comment:
Please add a comment describing why this is necessary.

Reply:
It is above, but I'll call it out.